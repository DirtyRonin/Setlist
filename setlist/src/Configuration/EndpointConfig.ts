import { ConfigurationItemCollection } from "./";

const WEB_API_URL = "http://localhost:5000/api";

export const defaultHeader = { "Access-Control-Allow-Origin": "*", "Content-Type": "application/json; charset=utf-8" };

export const GetEndpointURL = (endpointName: string): string => `${WEB_API_URL}/${endpointName}`;

type EndpointFieldNames = "Songs" | "Bandsongs" | "Bands";

type EndpointDefinition = {
    Name: string;
    GetEndpointUrl?: () => string;
    FilterUrl: string;
};

const EndpointPartialTypeDefinition: ConfigurationItemCollection<EndpointDefinition, EndpointFieldNames> = {
    Songs: { Name: "Songs", FilterUrl: "Filter" },
    Bandsongs: { Name: "Bandsongs", FilterUrl: "Filter" },
    Bands: { Name: "Bands", FilterUrl: "Filter" }
};

export const EndpointConfiguration: ConfigurationItemCollection<EndpointDefinition, EndpointFieldNames> = {
    Songs: {
        Name: EndpointPartialTypeDefinition.Songs.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Songs.Name),
        FilterUrl: EndpointPartialTypeDefinition.Songs.FilterUrl
    },
    Bandsongs: {
        Name: EndpointPartialTypeDefinition.Bandsongs.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Bandsongs.Name),
        FilterUrl: EndpointPartialTypeDefinition.Bandsongs.FilterUrl
    },
    Bands: {
        Name: EndpointPartialTypeDefinition.Bands.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Bands.Name),
        FilterUrl: EndpointPartialTypeDefinition.Bands.FilterUrl
    }
};
