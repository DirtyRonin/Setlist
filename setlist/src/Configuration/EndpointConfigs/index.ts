export * from "./EndPointConfig";
export * from "./BandSongsEndPointDefinition";
export * from "./SetlistSongsEndPointDefinition";
export * from "./SetlistsEndPointDefinition";
