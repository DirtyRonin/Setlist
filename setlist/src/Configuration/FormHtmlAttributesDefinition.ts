import { ConfigurationItemCollection } from "./ConfigurationItemCollection";

export type FormHtmlAttributesDefinition = {
    Label: string;
    Data_TestId: string;
    Placeholder: string;
    ControlId: string;
};

type AddSongToBandComponentNames = "ShowAddBandCheckBox" | "Navbar" | "NodeList"

export const AddSongToBandHtmlAttributesConfiguration: ConfigurationItemCollection<
    FormHtmlAttributesDefinition,
    AddSongToBandComponentNames
> = {
    ShowAddBandCheckBox: { Label: "Add Band", Placeholder: "", Data_TestId: "AddSongToBand_ShowAddBand", ControlId: "AddSongToBand_ShowAddBand" },
    Navbar: { Label: "", Placeholder: "", Data_TestId: "AddSongToBand_Options_responsiveNavbar", ControlId: "AddSongToBand_Options_responsiveNavbar" },
    NodeList: { Label: "", Placeholder: "", Data_TestId: "AddSongToBand_NodeList", ControlId: "AddSongToBand_NodeList" },
}

type AddSongToSetlistComponentNames = "ShowAddSetlistCheckBox" | "Navbar" | "NodeList"

export const AddSongToSetlistHtmlAttributesConfiguration: ConfigurationItemCollection<
    FormHtmlAttributesDefinition,
    AddSongToSetlistComponentNames
> = {
    ShowAddSetlistCheckBox: { Label: "Add Setlist", Placeholder: "", Data_TestId: "AddSongToSetlist_ShowAddSetlist", ControlId: "AddSongToSetlist_ShowAddSetlist" },
    Navbar: { Label: "", Placeholder: "", Data_TestId: "AddSongToSetlist_Options_responsiveNavbar", ControlId: "AddSongToSetlist_Options_responsiveNavbar" },
    NodeList: { Label: "", Placeholder: "", Data_TestId: "AddSongToSetlist_NodeList", ControlId: "AddSongToSetlist_NodeList" },
}

type AddBandSongToSetlistComponentNames = "ShowAddSetlistCheckBox" | "Navbar" | "NodeList"

export const AddBandSongToSetlistHtmlAttributesConfiguration: ConfigurationItemCollection<
    FormHtmlAttributesDefinition,
    AddBandSongToSetlistComponentNames
> = {
    ShowAddSetlistCheckBox: { Label: "Add Setlist", Placeholder: "", Data_TestId: "AddBandSongToSetlist_ShowAddSetlist", ControlId: "AddBandSongToSetlist_ShowAddSetlist" },
    Navbar: { Label: "", Placeholder: "", Data_TestId: "AddBandSongToSetlist_Options_responsiveNavbar", ControlId: "AddBandSongToSetlist_Options_responsiveNavbar" },
    NodeList: { Label: "", Placeholder: "", Data_TestId: "AddBandSongToSetlist_NodeList", ControlId: "AddBandSongToSetlist_NodeList" },
}

type AddBandSongFromSongsComponentNames = "ShowAddSetlistCheckBox" | "Navbar" | "NodeList"

export const AddBandSongFromSongsHtmlAttributesConfiguration: ConfigurationItemCollection<
    FormHtmlAttributesDefinition,
    AddBandSongFromSongsComponentNames
> = {
    ShowAddSetlistCheckBox: { Label: "Add Setlist", Placeholder: "", Data_TestId: "AddBandSongFromSongs_ShowAddSetlist", ControlId: "AddBandSongFromSongs_ShowAddSetlist" },
    Navbar: { Label: "", Placeholder: "", Data_TestId: "AddBandSongFromSongs_Options_responsiveNavbar", ControlId: "AddBandSongFromSongs_Options_responsiveNavbar" },
    NodeList: { Label: "", Placeholder: "", Data_TestId: "AddBandSongFromSongs_NodeList", ControlId: "AddBandSongFromSongs_NodeList" },
}