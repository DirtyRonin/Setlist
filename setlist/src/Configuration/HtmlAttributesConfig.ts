import { ConfigurationItemCollection } from "./";

type HtmlAttributesDefinition = {
    label: string;
    Data_TestId: string;
};

type FromHtmlAttributesDefinition = {
    label: string;
    Data_TestId: string;
    Placeholder: string;
    ControlId: string;
};

type CreateSongNodeHtmlComponentNames = "Title" | "Artist" | "Mode";

export const CreateSongNodeHtmlAttributesConfiguration: ConfigurationItemCollection<
    FromHtmlAttributesDefinition,
    CreateSongNodeHtmlComponentNames
> = {
    Title: { label: "Title", Placeholder: "Enter Title", Data_TestId: "CreateSongNode_Title", ControlId: "CreateSongNode_Title" },
    Artist: { label: "Artist", Placeholder: "Enter Artist", Data_TestId: "CreateSongNode_Artist", ControlId: "CreateSongNode_Artist" },
    Mode: { label: "Mode", Placeholder: "Enter Mode", Data_TestId: "CreateSongNode_Mode", ControlId: "CreateSongNode_Mode" }
};

type SongNodeHtmlComponentNames = CreateSongNodeHtmlComponentNames | "Id";

export const SongNodeHtmlAttributesConfiguration: ConfigurationItemCollection<HtmlAttributesDefinition, SongNodeHtmlComponentNames> = {
    Id: { label: "id", Data_TestId: "SongNode_Id" },
    Title: { label: CreateSongNodeHtmlAttributesConfiguration.Title.label, Data_TestId: "SongNode_Title" },
    Artist: { label: CreateSongNodeHtmlAttributesConfiguration.Artist.label, Data_TestId: "SongNode_Artist" },
    Mode: { label: CreateSongNodeHtmlAttributesConfiguration.Mode.label, Data_TestId: "SongNode_Mode" }
};

type CreateSetlistHtmlComponentNames = "SetlistNameInput" | "MajorLibraryNameInput" | "IsLibraryCheckbox" | "CreateSetlistButton"|"CreateMajorLibraryButton";

export const CreateSetlistHtmlAttributesConfiguration: ConfigurationItemCollection<
    FromHtmlAttributesDefinition,
    CreateSetlistHtmlComponentNames
> = {
    SetlistNameInput: { label: "Setlist Name", Placeholder: "Enter Setlist Name", Data_TestId: "CreateSetlist_SetlistName", ControlId: "CreateSetlist_SetlistName" },
    MajorLibraryNameInput: { label: "Major Library Name", Placeholder: "Enter Major Library Name", Data_TestId: "CreateSetlist_SetlistName", ControlId: "CreateSetlist_MajorLibraryName" },
    IsLibraryCheckbox: { label: "Library", Placeholder: "Select Setlist Attribute", Data_TestId: "CreateSetlist_IsLibrary", ControlId: "CreateSetlist_IsLibrary" },
    CreateSetlistButton: { label: "Create Setlist", Placeholder: "", Data_TestId: "CreateSetlist_CreateSetlist", ControlId: "CreateSetlist_CreateSetlist" },
    CreateMajorLibraryButton: { label: "Create Major Library", Placeholder: "", Data_TestId: "CreateSetlist_MajorLibrary", ControlId: "CreateSetlist_MajorLibrary" },
};


