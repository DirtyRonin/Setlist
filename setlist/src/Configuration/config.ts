type ConfigurationItemCollection<T, N extends string> = {
    [P in N]: T;
};

// Web Api
const WEB_API_URL = "http://localhost:5000/api";

export const ACCESS_CONTROL_ALLOW_ORIGIN_HEADER = { "Access-Control-Allow-Origin": "*" };

type EndpointFieldNames = "Songs" | "Setlists" | "Bands";

type EndpointDefinition = {
    Name: string;
    GetEndpointUrl?: () => string;
};

export const GetEndpointURL = (endpointName: string): string => `${WEB_API_URL}/${endpointName}`;

type SongFieldNames = "Id" | "Title" | "Artist" | "Mode";

type FieldDefinition = {
    label: string;
    Title: string;
    Placeholder: string;
    Data_TestId: string;
};

const EndpointPartialTypeDefinition: ConfigurationItemCollection<EndpointDefinition, EndpointFieldNames> = {
    Songs: { Name: "Songs" },
    Setlists: { Name: "Setlists" },
    Bands: { Name: "Bands" }
};

const EndpointTypeDefinition: ConfigurationItemCollection<EndpointDefinition, EndpointFieldNames> = {
    Songs: {
        Name: EndpointPartialTypeDefinition.Songs.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Songs.Name)
    },
    Setlists: {
        Name: EndpointPartialTypeDefinition.Setlists.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Setlists.Name)
    },
    Bands: {
        Name: EndpointPartialTypeDefinition.Bands.Name,
        GetEndpointUrl: () => GetEndpointURL(EndpointPartialTypeDefinition.Bands.Name)
    }
};

const SongTypeDefinition: ConfigurationItemCollection<FieldDefinition, SongFieldNames> = {
    Id: { label: "id", Title: "id", Placeholder: "",Data_TestId:"" },
    Title: { label: "Title", Title: "title", Placeholder: "Enter Title",Data_TestId:"SongNode_Title"  },
    Artist: { label: "Artist", Title: "artist", Placeholder: "Enter Artist",Data_TestId:"SongNode_Artist" },
    Mode: { label: "Mode", Title: "mode", Placeholder: "Enter Mode",Data_TestId:"SongNode_Mode" }
};

export default { SongTypeDefinition, EndpointTypeDefinition };
