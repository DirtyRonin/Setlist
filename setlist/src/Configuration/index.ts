export * from "./HtmlAttributesConfig"
export * from "./EndpointConfigs/index"