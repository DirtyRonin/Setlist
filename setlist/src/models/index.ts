import { ISongCatalog, ISetCatalog } from "./songCatalog"
import { IBandCatalog } from "./bandCatalog"

export * from "./actions/index"
export * from "./catalog"
export * from "./songCatalog"
export * from "./bandCatalog"
export * from "./song"
export * from "./band"
export * from "./modal"
export * from "./api"

export type Catalogs = ISongCatalog | IBandCatalog | ISetCatalog