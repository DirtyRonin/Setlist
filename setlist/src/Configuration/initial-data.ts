import { dndList } from "../models/DndListModels";

export const initialData: dndList = {
    songs: {
        "task-1": { id: "task-1", title: "Take out the garbage" , artist: "ArtistName", mode: "C"},
        "task-2": { id: "task-2", title: "Watch my favorite show", artist: "ArtistName", mode: "C" },
        "task-3": { id: "task-3", title: "charge my phone", artist: "ArtistName", mode: "C" },
        "task-4": { id: "task-4", title: "Cook dinner", artist: "ArtistName", mode: "C" }
    },
    setlists: {
        "column-1": {
            id: "column-1",
            title: "To do",
            songIds: ["task-1", "task-2", "task-3", "task-4"]
        },
        "column-2": {
            id: "column-2",
            title: "Finished",
            songIds: []
        }
    },
    setlistOrder: ["column-1", "column-2"]
};

