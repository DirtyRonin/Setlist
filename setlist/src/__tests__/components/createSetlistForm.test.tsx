import React from "react";
import { render, within, waitForElement, fireEvent, act, getByLabelText, queryByAttribute } from "@testing-library/react";

import CreateSetlist, { ICreateSetlistProps } from "../../components/createSetlistForm"
import { CreateSetlistHtmlAttributesConfiguration } from "../../Configuration";
import { IApiBandlist } from "../../models/";

const htmlConfig = CreateSetlistHtmlAttributesConfiguration;

const mockCreateBandAsync = jest.fn().mockResolvedValue({})
const mockAddBandToState = jest.fn()

const props: ICreateSetlistProps = {
    IsBandListNeeded: true,
    CreateBandAsync: (setlist: IApiBandlist): Promise<IApiBandlist> => Promise.resolve({ ...setlist, id: "newId" }),
    AddBandToState: (setlist: IApiBandlist): void => { }
}

const renderCreateSetlistForm = async (defaultprops: ICreateSetlistProps, props: Partial<ICreateSetlistProps> = {}) =>
    render(<CreateSetlist {...defaultprops} {...props} />)

describe("Test create setlist form", () => {

    beforeEach(() => {
        mockCreateBandAsync.mockClear();
        mockAddBandToState.mockClear();
    })

    it("Should show the setlistname input when ismajorLibrary is true", async () => {
        const { asFragment } = await renderCreateSetlistForm(props);
        expect(asFragment()).toMatchSnapshot();
    })

    it("Should call the async create setlist function with expectedParams when ismajorlibrary is true", async () => {

        const newProps: ICreateSetlistProps = {
            ...props,
            CreateBandAsync: mockCreateBandAsync
        }

        const { getByPlaceholderText, getByText } = await renderCreateSetlistForm(newProps);

        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.MajorLibraryNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Major Lazor" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.MajorLibraryNameInput.Placeholder));

        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateMajorLibraryButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateMajorLibraryButton.label));

        const expectedParams: IApiBandlist = {
            id: "",
            title: "Major Lazor",
            bandsongs: []
        }

        expect(mockCreateBandAsync).toHaveBeenCalledWith(expectedParams);
    })

    it("Should call the add setlist to state function with expectedParams when ismajorlibrary is true", async () => {

        const newProps: ICreateSetlistProps = {
            ...props,
            AddBandToState: mockAddBandToState
        }

        const { getByPlaceholderText, getByText } = await renderCreateSetlistForm(newProps);

        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.MajorLibraryNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Major Lazor" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.MajorLibraryNameInput.Placeholder));

        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateMajorLibraryButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateMajorLibraryButton.label));

        const expectedParams: IApiBandlist = {
            id: "newId",
            title: "Major Lazor",
            bandsongs: []
        }

        expect(mockAddBandToState).toHaveBeenCalledWith(expectedParams);
    })

    it("Should show the setlistname input and the libraray checkbox when ismajorLibrary is false", async () => {
        const newProps: ICreateSetlistProps = {
            ...props,
            IsBandListNeeded: false
        }

        const { asFragment } = await renderCreateSetlistForm(newProps);
        expect(asFragment()).toMatchSnapshot();
    })

    it("Should call the async create setlist function with expectedParams when ismajorlibrary is false", async () => {

        const newProps: ICreateSetlistProps = {
            ...props,
            IsBandListNeeded: false,
            CreateBandAsync: mockCreateBandAsync
        }

        const { getByPlaceholderText, getByText } = await renderCreateSetlistForm(newProps);
        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Rock Classics" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder));

        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateSetlistButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateSetlistButton.label));

        const expectedParams: IApiBandlist = {
            id: "",
            title: "Rock Classics",
            bandsongs: []
        }

        expect(mockCreateBandAsync).toHaveBeenCalledWith(expectedParams);
    })
    it("Should call the async create setlist function as a library with expectedParams when ismajorlibrary is false", async () => {

        const newProps: ICreateSetlistProps = {
            ...props,
            IsBandListNeeded: false,
            CreateBandAsync: mockCreateBandAsync
        }

        const {getByPlaceholderText,getByLabelText,getByText} = await renderCreateSetlistForm(newProps);
        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Rock Classics" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder));

        //toggle is library to true
        const isLibraryCheckbox = getByLabelText(htmlConfig.IsLibraryCheckbox.label)
        act(() => { fireEvent.click (isLibraryCheckbox); })
        await waitForElement(() => getByLabelText(htmlConfig.IsLibraryCheckbox.label));
       
        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateSetlistButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateSetlistButton.label));

        const expectedParams: IApiBandlist = {
            id: "",
            title: "Rock Classics",
            bandsongs: []
        }

        expect(mockCreateBandAsync).toHaveBeenCalledWith(expectedParams);
    })

    it("Should call the add setlist to state function with expectedParams when ismajorlibrary is false", async () => {
        const newProps: ICreateSetlistProps = {
            ...props,
            IsBandListNeeded: false,
            AddBandToState: mockAddBandToState
        }

        const { getByPlaceholderText, getByText } = await renderCreateSetlistForm(newProps);
        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Rock Classics" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder));

        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateSetlistButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateSetlistButton.label));

        const expectedParams: IApiBandlist = {
            id: "newId",
            title: "Rock Classics",
            bandsongs: []
        }

        expect(mockAddBandToState).toHaveBeenCalledWith(expectedParams);
    })

    it("Should call the add setlist to state function as a library with expectedParams when ismajorlibrary is false", async () => {
        const newProps: ICreateSetlistProps = {
            ...props,
            IsBandListNeeded: false,
            AddBandToState: mockAddBandToState
        }

        const { getByPlaceholderText, getByText ,getByLabelText} = await renderCreateSetlistForm(newProps);
        //enter name
        const nameInput = getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder)
        act(() => { fireEvent.change(nameInput, { target: { value: "Rock Classics" } }); })
        await waitForElement(() => getByPlaceholderText(htmlConfig.SetlistNameInput.Placeholder));

         //toggle is library to true
         const isLibraryCheckbox = getByLabelText(htmlConfig.IsLibraryCheckbox.label)
         act(() => { fireEvent.click (isLibraryCheckbox); })
         await waitForElement(() => getByLabelText(htmlConfig.IsLibraryCheckbox.label));

        //push create button
        const createMajorLibrary = getByText(htmlConfig.CreateSetlistButton.label)
        act(() => { fireEvent.click(createMajorLibrary); })
        await waitForElement(() => getByText(htmlConfig.CreateSetlistButton.label));

        const expectedParams: IApiBandlist = {
            id: "newId",
            title: "Rock Classics",
            bandsongs: []
        }

        expect(mockAddBandToState).toHaveBeenCalledWith(expectedParams);
    })

})