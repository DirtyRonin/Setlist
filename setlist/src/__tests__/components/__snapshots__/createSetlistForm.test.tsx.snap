// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test create setlist form Should show the setlistname input and the libraray checkbox when ismajorLibrary is false 1`] = `
<DocumentFragment>
  <form
    class=""
    method="GET"
  >
    <div
      class="form-group col-md-6"
    >
      <label
        class="form-label"
        for="CreateSetlist_SetlistName"
      >
        Setlist Name
      </label>
      <input
        class="form-control"
        id="CreateSetlist_SetlistName"
        placeholder="Enter Setlist Name"
        type="text"
      />
    </div>
    <div
      class="form-group col-md-6"
    >
      <div
        class="form-check"
      >
        <input
          class="form-check-input"
          id="CreateSetlist_IsLibrary"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="CreateSetlist_IsLibrary"
          title=""
        >
          Library
        </label>
      </div>
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Create Setlist
    </button>
  </form>
</DocumentFragment>
`;

exports[`Test create setlist form Should show the setlistname input when ismajorLibrary is true 1`] = `
<DocumentFragment>
  <form
    class=""
    method="GET"
  >
    <div
      class="form-group col-md-6"
    >
      <label
        class="form-label"
        for="CreateSetlist_MajorLibraryName"
      >
        Major Library Name
      </label>
      <input
        class="form-control"
        id="CreateSetlist_MajorLibraryName"
        placeholder="Enter Major Library Name"
        type="text"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Create Major Library
    </button>
  </form>
</DocumentFragment>
`;
